src/mcnntunes/minimizer.py

```diff
@@ -30,37 +32,65 @@ def __init__(self, runs, truth, model, output = None, truth_index = 0):
             self.write_on_disk = True
             self.output = output
 
+        # Compute degrees of freedom for chi2
+        self.dof = np.sum(self.runs.y_weight) - len(self.runs.params)
```

Comment:
I was thinking but if the weights sum is not an integer? That would be a problem for the chi2 distribution with self.dof degree of freedom?

Reply:
I believe it should work anyway with non-integer degrees of freedom.